tpblite/models/utils.py

```diff
@@ -1,57 +1,80 @@
+from typing import Tuple, Type
 import random
 from urllib.request import Request, urlopen
 import urllib.error
 from purl import URL as pURL
 
 
-class QueryParser(object):
-    '''
-    Query object capable of getting html response given 
-    a search query and other parameters.
-    '''
-    def __init__(self, query, base_url, page, order, category):
+class QueryParser:
+    """Query object capable of getting html response given a search query and other
+    parameters.
+    """
+
+    # PirateBay URL to use for queries
+    base_url: str
+
+    # Compiled search string used to query the PirateBay URL
+    url: str
+
+    def __init__(self, base_url: str, segments: Tuple[str, ...]):
```

Comment:
I chose not to use type hints in this codebase as I think it potentially can look a bit messy, however if you think it is beneficial at this point to include them then i would be happy to do so. What's your opinion on this?

I would ask that if they're going to be included that they should be consistently across the codebase, including the Torrent and Torrents objects.

Reply:
> I chose not to use type hints in this codebase as I think it potentially can look a bit messy, however if you think it is beneficial at this point to include them then i would be happy to do so. What's your opinion on this?

I'm probably a bit biased but I **love** type hinting :). We use it everywhere at work and I really see huge benefits:

- it's clearer: here for example, you could wonder if base url is a string or some fancy URL object. The type hinting clarifies the situation. I think this alone is already worth it

- The real gain is when you use a typed codebase with mypy. Mypy is some sort of linter/tool that will catch bugs before you run your code. Let's say for example that in another project I import the `QueryParser` object, and I try to do something like this:

```
import QueryParser
q = QueryParser()
q.base_url + 5
```

It's a dummy example obviously, but you can see this code would crash at runtime because you can't add an int to a str. With mypy as a linter, you would catch this bug *before* running the code. Here it doesn't seem ground breaking, but on a big codebase, this is a game changer IMO.

Also the type hinting doesn't impact the code execution *at all*, this is just an aid to development. You can do it gradually on some parts of the code only. I've seen some codebases half-typed, where just the functions that communicate with the "outside" are typed.

Reply:
Okay, you've convinced me, happy to have type hints!

Reply:
Yes! hahah :)